### PR TITLE
Update doc for feature constraints.  Fix some warnings.

### PR DESCRIPTION
**Pull request by trivialfis**
I'm not confident about my writing, please help verifying.  Otherwise it might be very embarrassing ...

I am not sure about how to fix the following warnings yet:
```
/home/fis/Workspace/xgb/xgboost/doc/gpu/index.rst:89: WARNING: Inline interpreted text or phrase reference start-string without end-string.
/home/fis/Workspace/xgb/xgboost/doc/jvm/index.rst:3: WARNING: Duplicate explicit target name: "here".
/home/fis/Workspace/xgb/xgboost/doc/jvm/xgboost4j_spark_tutorial.rst:3: WARNING: Duplicate explicit target name: "here".
/home/fis/Workspace/xgb/xgboost/doc/jvm/xgboost4j_spark_tutorial.rst:3: WARNING: Duplicate explicit target name: "here".
/home/fis/Workspace/xgb/xgboost/python-package/xgboost/plotting.py:docstring of xgboost.to_graphviz:19: WARNING: Unexpected indentation.
/home/fis/Workspace/xgb/xgboost/python-package/xgboost/plotting.py:docstring of xgboost.to_graphviz:24: WARNING: Unexpected indentation.

```

### CODE REVIEW

File: python-package/xgboost/dask.py [hunk 1]
```diff
@@ -101,24 +101,25 @@ def _run_with_rabit(rabit_args, func, *args):
...
+    xgb.train() will use the threads allocated by dask by default, unless the
+    user overrides the nthread parameter.
 
-    Note: Windows platforms are not officially supported. Contributions are welcome here.
+    Note: Windows platforms are not officially
+      supported. Contributions are welcome here.
 
     :param client: Dask client representing the cluster
-    :param func: Python function to be executed by each worker. Typically contains xgboost
-    training code.
```
Comment (trivialfis) [inline python-package/xgboost/dask.py]: This change corresponds to a warning about `unexpected indent`.

File: python-package/xgboost/plotting.py [hunk 2]
```diff
@@ -184,18 +184,16 @@ def to_graphviz(booster, fmap='', num_trees=0, rankdir='UT',
     no_color : str, default '#FF0000'
         Edge color when doesn't meet the node condition.
     condition_node_params : dict (optional)
-        condition node configuration,
```
Comment (trivialfis) [inline python-package/xgboost/plotting.py]: Corresponds to unexpected indent and something like code block ending without a new line.. The former is still here.

File: python-package/xgboost/plotting.py [hunk 3]
```diff
@@ -184,18 +184,16 @@ def to_graphviz(booster, fmap='', num_trees=0, rankdir='UT',
...
-        {'shape':'box',
-               'style':'filled,rounded',
```
Comment (trivialfis) [inline python-package/xgboost/plotting.py]: Unexpected indent.

File: python-package/xgboost/sklearn.py [hunk 4]
```diff
@@ -105,8 +105,8 @@ class XGBModel(XGBModelBase):
         Value in the data which needs to be present as a missing value. If
         None, defaults to np.nan.
     importance_type: string, default "gain"
-        The feature importance type for the feature_importances_ property: either "gain",
```
Comment (trivialfis) [inline python-package/xgboost/sklearn.py]: The under scope is interpreted as some sphinx inlined target.

File: doc/gpu/index.rst [hunk 5]
```diff
@@ -101,9 +100,6 @@ Objective functions
 ===================
 Most of the objective functions implemented in XGBoost can be run on GPU.  Following table shows current support status.
 
-.. |tick| unicode:: U+2714
```
Comment (trivialfis) [inline doc/gpu/index.rst]: This change corresponds to warning about duplicated unicode definition.

**Comment by trivialfis** [comment]
@RAMitchell One of the example is borrowed from you. :) Thanks.

File: doc/gpu/index.rst [hunk 6]
```diff
@@ -86,8 +86,7 @@ Multiple GPUs can be used with the ``gpu_hist`` tree method using the ``n_gpus``
 
 .. note:: Enabling multi-GPU training
 
-  Default installation may not enable multi-GPU training. To use multiple GPUs, make sure to read :ref:`build_gpu_support`.
-XGBoost supports multi-GPU training on a single machine via specifying the `n_gpus' parameter.
+  Default installation may not enable multi-GPU training. To use multiple GPUs, make sure to read :ref:`build_gpu_support`.  XGBoost supports multi-GPU training on a single machine via specifying the `n_gpus' parameter.
```
Comment (RAMitchell) [inline doc/gpu/index.rst]: `XGBoost supports multi-GPU training on a single machine via specifying the `n_gpus' parameter.` This sentence can be removed as this is no longer true.

**Pull request merged**